modules/standard/Socket.chpl

```diff
@@ -0,0 +1,1403 @@
+  }
+  var localSync$: sync c_short;
+  // create event pending state
+  var internalEvent = event_new(event_loop_base, this.socketFd, EV_READ | EV_TIMEOUT, c_ptrTo(syncRWTCallback), c_ptrTo(localSync$):c_void_ptr);
+  defer {
+    // cleanup
+    event_free(internalEvent);
+  }
+
+  while err_out != 0 {
+    var t: Timer;
+    t.start();
+    // make event active
+    if timeout.tv_sec == -1 {
+      err_out = event_add(internalEvent, nil);
+    }
+    else {
+      err_out = event_add(internalEvent, c_ptrTo(timeout));
+    }
+    if err_out != 0 {
+      throw new Error("accept() failed");
+    }
+    // return value
+    var retval = localSync$.readFE();
+    // stop timer
+    t.stop();
+    // if error was timeout throw error
+    if retval & EV_TIMEOUT != 0 {
+      throw SystemError.fromSyserr(ETIMEDOUT, "accept() timed out");
+    }
+    var elapsedTime = t.elapsed(TimeUnits.microseconds):c_long;
+    // try accept again
+    err_out = sys_accept(socketFd, client_addr, fdOut);
+    if err_out != 0 {
+      // error was not about blocking wait so throw it
+      if err_out != EAGAIN && err_out != EWOULDBLOCK {
+        throw SystemError.fromSyserr(err_out, "accept() failed");
+      }
+      // no indefinitely blocking wait
+      if timeout.tv_sec != -1 {
+        var totalTimeout = timeout.tv_sec*1000000 + timeout.tv_usec;
+        // timer didn't elapsed
+        if totalTimeout > t.elapsed(TimeUnits.microseconds) {
+          const remainingMicroSeconds = ((totalTimeout - elapsedTime)%1000000);
+          const remainingSeconds = ((totalTimeout - elapsedTime)/1000000);
+          timeout.tv_sec = remainingSeconds;
+          timeout.tv_usec = remainingMicroSeconds;
+        }
+        throw SystemError.fromSyserr(ETIMEDOUT, "accept() timed out");
+      }
+    }
+  }
+  return openfd(fdOut):tcpConn;
+}
+
+/*
+  Close the file descriptor
+*/
+proc ref tcpListener.close() throws {
+  var err_out = sys_close(this.socketFd);
+  if err_out != 0 {
+    throw SystemError.fromSyserr(err_out, "Failed to close tcpListener");
+  }
+  this.socketFd = -1;
+}
+
+/*
+  Returns the address on which socket is
+  listening on and bound to.
+
+  :return: bound address
+  :rtype: `ipAddr`
+*/
+proc tcpListener.addr throws {
+  return getsockname(this.socketFd);
+}
+
+inline operator !=(const ref lhs: tcpListener,const ref rhs: tcpListener) {
+  return lhs.socketFd != rhs.socketFd;
+}
+
+inline operator ==(const ref lhs: tcpListener,const ref rhs: tcpListener) {
+  return !(lhs.socketFd != rhs.socketFd);
+}
+
+proc tcpListener.writeThis(f) throws {
+  f.write("(","addr:",this.addr,",fd:",this.socketFd);
+}
+
+pragma "no doc"
+extern const SOMAXCONN: int;
+/*
+  Default `backlog` value used in :proc:`listen`
+  It is calulated as min(`SOMAXCONN`, 128) where `SOMAXCONN` is
+  the maximum number of allowed pending connections in the system.
+*/
+var backlogDefault:uint(16) = (if SOMAXCONN <= 128 then SOMAXCONN else 128):uint(16);
```

Comment:
Is this something that can change? Or be changed by the user?

Consider making it a `const` and/or `no doc`.